tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs

```diff
@@ -1219,4 +1224,19 @@ let ``AsyncSeq.groupBy should propagate exception and terminate all groups``() =
     asyncSeq { raise (exn("test")) } 
     |> AsyncSeq.groupBy (fun i -> i % 3) 
     |> AsyncSeq.mapAsyncParallel (snd >> AsyncSeq.toListAsync)
-  Assert.AreEqual(expected, actual)
\ No newline at end of file
+  Assert.AreEqual(expected, actual)
+
+[<Test>]
+let ``AsyncSeq.combineLatest should behave like merge after initial``() =  
+  for n in 0..20 do
+    for m in 0..10 do
+      let ls1 = List.init n id
+      let ls2 = List.init m id 
+      // expect each element to increase combined sum by 1
+      // expected count is sum of source counts minus 1 for first result
+      let expectedCount = 
+        if n = 0 || m = 0 then 0
+        else max (n + m - 1) 0
+      let expected = List.init expectedCount id |> AsyncSeq.ofSeq
```

Comment:
I pushed an additional test to check for a timeout. Is this what you had in mind?